[TOPIC-BLE-LLCP] Bluetooth: controller: inclusive naming and added includes

kruithofa opened the pull request:
Some inclusive naming changes were already in the main branch, which resulted in compile errors
Some includes were missing, causing failures in CI
Some EDTT tests are failing, disabled pending investigation

Signed-off-by: Andries Kruithof <Andries.Kruithof@nordicsemi.no>

Commit:
Bluetooth: controller: update after sync in September

Update is required due to name change from MASTER/SLAVE
to CENTRAL/PERIPHERAL

This allows unit tests to compile and run

Signed-off-by: Andries Kruithof <Andries.Kruithof@nordicsemi.no>

Commit:
Bluetooth: controller: allow direction finding samples to compile

This fixes missing includes for the direction finding samples:
when including ull_internal.h we must also include lll_conn.h for
getting the type definition of the node_tx struct.
Without this change the samples direction_finding_connectionless_rx
and direction_finding_connectionless_tx will fail
This change is required in topic branch only, not in main

Also adds missing conditional compiles,i
which caused build errors for babblesim

This replaces PR 38067

Signed-off-by: Andries Kruithof <Andries.Kruithof@nordicsemi.no>

Commit:
Bluetooth: controller: disable failing EDTT tests

Disabled some EDTT tests, mainly related to DLE
pending further investigation

Signed-off-by: Andries Kruithof <Andries.Kruithof@nordicsemi.no>

erbr-ot commented on subsys/bluetooth/controller/ll_sw/ull_chan.c:
```diff
@@ -38,8 +39,11 @@
 static uint8_t count = 37U;
 
+#if defined(CONFIG_BT_LL_SW_SPLIT_LLCP_LEGACY)
 static void chan_map_set(uint8_t const *const chan_map);
```
Why not remove the static version in the legacy build and simply use ull_chan_map_set?

erbr-ot reviewed (approved):
Only issue I have is re. chan_map_set()

thoh-ot commented:
Wan't CI supposed to be green?

Commit:
Bluetooth: controller: rename variable time

'time' is a reserved identifier, and can not be used
as a variable name. Renamed the 2 occurences to avoid CI failure

Signed-off-by: Andries Kruithof <Andries.Kruithof@nordicsemi.no>

kruithofa commented:
> Wasn't CI supposed to be green?

That was the expectation yes

Commit:
Bluetooth: controller: fix linelengths to be below 100

Some lines were over 100 characters, which causes CI failure

Signed-off-by: Andries Kruithof <Andries.Kruithof@nordicsemi.no>

Commit:
Bluetooth: controller: remove unused variables, add missing include

This will fix some of the CI failures

Signed-off-by: Andries Kruithof <Andries.Kruithof@nordicsemi.no>

kruithofa commented:
So let's see if things are improved now

Commit:
Bluetooth: controller: fix typo in kconfig names

Fix a typo in some kconfig names

Signed-off-by: Andries Kruithof <Andries.Kruithof@nordicsemi.no>

erbr-ot reviewed (approved):
Still the question re. chan_map_set() but that apart it looks just dandy to me
